### DISCUSSION TITLE
postgres_fdw -> Crate

### DISCUSSION THREAD

**Discussion by robd003** [crate/crate#12581] [Q&A] [answered]
Has anyone had luck connecting Crate to a Postgres server via [foreign data wrappers](https://www.postgresql.org/docs/current/postgres-fdw.html)?

**Comment by proddata** [comment]
We tested it a while ago. As far as I remember one missing feature was [cursor support](https://github.com/crate/crate/issues/10212), which currently is being worked on and might be released with 5.0 within the coming weeks, or otherwise the next iteration.

**Comment by robd003** [comment]
By the way it would be _**AMAZING**_ if Crate could do it's own "FDW" connections to other Postgres databases.

Being able to join from external postgres databases would be immensely useful. At the moment I've been writing crappy functions that will annotate the results from Crate with extra properties from other Postgres databases. (I tried doing nightly database sync's but we'd always have missing rows throughout the day)

...just posted this and and then saw that it's on the list https://github.com/crate/crate/issues/12686

Crate rocks!

**Comment by hlcianfagna** [comment, marked as answer]
Hi @robd003 ,
just wanted to give you a quick update on this front, 

since CrateDB added support for `DECLARE CURSOR` it is now possible to use `postgres_fdw` to query CrateDB from a PostgreSQL instance (but there are currently some limitations depending on the queries and the version of PostgreSQL - see https://community.crate.io/t/cratedb-postgres-fdw-limit-parameter/1420 for details)

you also mention syncing of data from other DBs to CrateDB
while waiting for https://github.com/crate/crate/issues/12686
you could use the approach from https://community.crate.io/t/replicating-data-from-other-databases-to-cratedb-with-debezium-and-kafka/1388

I hope this helps.
